### ISSUE TITLE
Content not well explained

### ISSUE THREAD

**Issue by ScottDonnan** [learn-co-curriculum/phase-4-lecture-videos-client-server-post-request-error-handling#1]
### Canvas Link

https://learning.flatironschool.com/courses/4379/pages/video-client-server-post-request-error-handling?module_item_id=318729

### Concern

It's possible i've missed some material, but upon seeing the line of code in line 37 I do not feel that all the pieces of this error handling were explained.  Specifically what do all the pieces of this code do?

res.json().then(e => setErrors(Object.entries(e.error).flat()))

we have covered res.json(), and setting state with the setter function (setErrors) but have had very very exposure to the other pieces.  Is objects a Class? havn't talked about that in javascript.  what are ".entries"? and .flat()? 

Basically it feels not well explained.  Hopefully this is not too confusing.  Thank you.

### Suggested Changes

Talk in more detail about these last pieces of code in the error handling since we have not been exposed to them before.
